images/virtualization-artifact/pkg/controller/service/disk_service.go

```diff
@@ -509,12 +510,26 @@ func (s DiskService) getDefaultStorageClass(ctx context.Context) (*storev1.Stora
 		return nil, err
 	}
 
+	defaultClasses := []storev1.StorageClass{}
 	for _, sc := range scs.Items {
 		if sc.Annotations[common.AnnDefaultStorageClass] == "true" {
-			return &sc, nil
+			defaultClasses = append(defaultClasses, sc)
 		}
 	}
 
+	if len(defaultClasses) == 0 {
+		return nil, ErrDefaultStorageClassNotFound
+	}
+
+	// Primary sort by creation timestamp, newest first
+	// Secondary sort by class name, ascending order
+	sort.Slice(defaultClasses, func(i, j int) bool {
+		if defaultClasses[i].CreationTimestamp.UnixNano() == defaultClasses[j].CreationTimestamp.UnixNano() {
+			return defaultClasses[i].Name < defaultClasses[j].Name
+		}
+		return defaultClasses[i].CreationTimestamp.UnixNano() > defaultClasses[j].CreationTimestamp.UnixNano()
+	})
+
 	return nil, ErrDefaultStorageClassNotFound
```

Comment:
Always nil return, but default storage class was found
```suggestion
	return defaultClasses[0], nil
```